### DISCUSSION TITLE
[Fair request?] Add "navigate" attribute in ActionCreateTiddlerWidget

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5531] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is unclear how to directly navigate to a tiddler created with the ActionCreateTiddlerWidget because the new *title* of the created tiddler is dynamically numbered.

**Describe the solution you'd like**
I wish for a `navigate` attribute. If this attribute is set to `navigate=edit` or `navigate=editmode`, one is navigated to the edit view of the tiddler. All other arguments navigate to the viewmode of the tiddler. 

The idea to avoid an explicit argument for the latter is because it is difficult to remember it: *"What is the opposide of *edit view*? "View view?"* We do use "edit *mode*" and "view *mode*" but I propose that anything should go e.g `=view` `=viewmode` `="view mode"` `=yes` `=true`. (Maybe even no argument i.e just the attrbute `navigate`.)

**Comment by pmario** [comment]
Hmmm,

At the moment ActionCreateTiddlerWidget is the widget, that doesn't navigate by design. I did add new functions to action-create, since it didn't have the same features as tm-new-tiddler. 

ActionSendMessage $message="tm-new-tiddler / tm-edit-tiddler" is the action widget, that does navigate. see: https://tiddlywiki.com/#ActionSendMessageWidget

**Reply by twMat** [reply]
>[...] doesn't navigate by design. I did add new functions to action-create, since it didn't have the same features as tm-new-tiddler.

Where did you add these new functions? (That does indicate you agree the request makes sense then.)

>ActionSendMessage $message="tm-new-tiddler / tm-edit-tiddler" is the action widget, that does navigate. see: https://tiddlywiki.com/#ActionSendMessageWidget

How actually? To the *unknown* new title? E.g "New tiddler **2**"  - Regardless, most people in the TW community would probably call me a TW expert after 15 years of fiddling. Yet, I don't know how to do this. I probably *have* done it some time in the past but, like now, it was surely *way* too demanding and time consuming. So my proposal is to make it practical. IMO this attribute should be a given but since you say it was intentionally left out, do you recall why?

Thanks for your input!

**Reply by pmario** [reply]
The default behavior of the `tm-new-tiddler` does the exact same thing as the default behavior of the action-createtiddler widget. The only difference is the navigation part.

```
<$button>
<$action-sendmessage $message="tm-new-tiddler" />
New Tiddler
</$button>
```

**Comment by pmario** [comment]
Action-widgets have been created, because the button-widget was completely overloaded with features and parameters. It got more and more complicated.  IMO in the end it was hard to use for "standard users".

Action widgets are designed to do "a few" things well, so they are easy to combine to do more complex things.

**Reply by saqimtiaz** [reply]
I believe you meant "a few" rather than "a view" @pmario?

Yes the idea with action widgets is to keep their focus narrow and make it easy to combine multiple widgets.

Also sometimes overlooked in the context of generating titles is the `unusedtitle` macro.

**Reply by twMat** [reply]
I appreciate your input, guys. Maybe my OP stems from insufficient [docs](https://tiddlywiki.com/#WidgetMessage%3A%20tm-new-tiddler) on the matter ( *- What is "the `event` object"? Why do they call it "Name" instad of "Attribute"? Why is it called "param" when it should be called "template"?* etc, to just name confusions from the first few rows in that doc tiddler.)

Or this bit that you mention Saq:

>Also sometimes overlooked in the context of generating titles is the `unusedtitle` macro.

This is no detail, and it is not mentioned in the docs.

**Reply by saqimtiaz** [reply]
https://tiddlywiki.com/#unusedtitle%20Macro

You can use this to predict/precalculate the title that will be used by ActionCreateTiddler.

**Reply by saqimtiaz** [reply]
For tm-new-tiddler the first few paragraphs are written from a developer point of view.

The examples at the end of that tiddler are what you want, especially the one using action-sendmessage.

**Reply by pmario** [reply]
There is a pending PR: **allow unusedtitle macro to use the prefix parameter and fix wiki.generateNewTitle() #5361**

It contains a link to a working version of the new macro.
